commit 5775da00c7b4e785163e9ae9d5499cb7601f719c
Author: bertin <macportsuser@macports.org>
Date:   Fri May 23 21:30:19 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts-pextlib/files/patch-better-registry-error.diff

diff --git a/src/registry2.0/registry_util.tcl b/src/registry2.0/registry_util.tcl
index fdb4a5de..509c2a73 100644
--- a/src/registry2.0/registry_util.tcl
+++ b/src/registry2.0/registry_util.tcl
@@ -131,7 +131,7 @@ proc run_target {port target options} {
         return 0
     }
 
-    if {![catch {set mport [mportopen_installed [$port name] [$port version] [$port revision] [$port variants] $options]}]} {
+    if {![catch {set mport [mportopen_installed [$port name] [$port version] [$port revision] [$port variants] $options]} err]} {
         set failed 0
         if {[catch {mportexec $mport $target} result]} {
             ui_debug $::errorInfo
@@ -169,7 +169,7 @@ proc run_target {port target options} {
         }
     } else {
         ui_debug $::errorInfo
-        ui_warn "Failed to open Portfile from registry for $portspec"
+        ui_warn "Failed to open Portfile from registry for $portspec ($err); registry=[$port portfile]"
     }
     return 0
 }
